src/Helper/Helper.php

```diff
@@ -68,18 +93,198 @@ public function getSubmissionIdFromJob(int $jobId): ?int {
   }
 
   /**
-   * Get all jobs from advanced queue table.
+   * Get all jobs that match a specific form.
+   *
+   * @param string $formId
+   *   The form to match.
    *
    * @return array
-   *   A list of jobs.
+   *   A list of view parameters.
    *
-   * @phpstan-return array<int, \stdClass>
+   * @phpstan-return array<string, mixed>
    */
-  public function getAllJobs():array {
-    $query = $this->connection->select('advancedqueue', 'a');
-    $query->fields('a', ['payload', 'job_id']);
+  public function getQueueJobIds(string $formId): array {
+    $query = $this->connection->select('os2forms_queue_submission_relation', 'o');
+    $query->fields('o', ['job_id']);
+    $query->condition('webform_id', $formId, '=');
+
+    return $query->execute()->fetchCol();
+  }
+
+  /**
+   * Handle a job from the AdvancedQueueProcessSubscriber.
+   *
+   * @param \Drupal\advancedqueue\Job $job
+   *   The job about to be processed.
+   */
+  public function handleJob(Job $job): void {
+    $data = $this->getDataFromPayload($job->getPayload());
+    if ($data) {
+      $data['job_id'] = (int) $job->getId();
+      $this->addRelation($data);
+    }
```

Comment:
I think the code will be a little cleaner if you rename `getDataFromPayload` to `getDataFromJob`, pass it the `Job` and have it return the full data (including `job_id`). 

Consider removing the `addRelation` function and move it's body into this function's body.  `addRelation` is only used in one place and is not too complex so putting the logic into `handleJob` will not make the code more complex.

Reply:
Moved